GoogleForCreators/web-stories-wp
# Editor: Enforce minimum and maximum values for input fields

## Summary

This PR improve the way that we are currently enforcing min/max values on inputs.

## Relevant Technical Choices

- Defined a helper to process and return values between min/max ranges specced. 
- Defined a helper to validate min/max ranges specced from a specific source (e.g. mostly `.length` property).
- Used local constants to define by scope each min/max field value
- Used `usePresubmitHandler` when possible to reflect the current UX and data flow.
- Used `min`, `max`, `minLength`, and `maxLength` attributes when possible to improve code quality for web standards once we have specced min/max values for some inputs and for further usage when we define the UX/UI feedback for those min/max values.
- `opacity` received a special improvement since it was not being validated correctly considering `1` the minimum, but now, `0` is the minimum.

## User-facing changes
 
Inputs covered on the initial PR will reflect limitations as soon as the user input some wrong value. 

- Numeric fields will update using the closest range (e.g. Lower values of the limits will use the min allowed, higher values of limits will use the max allowed)
- Text fields will limit the user to input more than the max allowed. Also, text fields only will execute scoped actions after user inputs reach the min values required.

## To-Do

- Add more tests

## Testing Instructions

- Add an image element
- Select the dimensions (width and/or height) field in the Design Panel
- Type `10000` and press Enter
- It should update the field selected (or both if aspect-ratio is locked) to `1000`

---



Solves #1549
Solves #1881 
Solves #1419
